### PR TITLE
Feat : add Cart and Wishlist

### PR DESCRIPTION
**Pull request by ZuberDunge**

### CODE REVIEW

File: src/frontend/context/cartContext.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+const CartContextProvider = (props) => {
+    const [cart, setCart] = useState([])
+
+    const deleteItem = (id) => {
+        setCart(cart.filter(cart => cart._id !== id))
+    }
+
+    let token = localStorage.getItem("userToken")
+    console.log(token);
```
Comment (Sameony) [inline src/frontend/context/cartContext.js:13]: you can remove the console messages after you've confirmed the functionality

File: src/frontend/pages/cart.jsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    return (<>
+        {cart.length == 0 || cart == null ? <div className="empty-cart">
+            Add something here
+        </div> :
+            <div className="cart">
+                <div className="cart-grid">
+                    <div className="cart-items">
+
+                        <div
+                            className="bat-flex wrap-on-500 bat-align-center flex-gap-1 bat-pad-1 bat-justify-between"
```
Comment (Sameony) [inline src/frontend/pages/cart.jsx:43]: great naming and use of utility classes

File: src/frontend/style/cart.css [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    margin: 0 1rem;
+}
+
+.cart-add-qnt span:nth-child(2) {
+    border: 1px solid var(--grey);
+    height: 1.5rem;
+    width: 3rem;
+    display: flex;
+    justify-content: center;
+    align-items: center;
```
Comment (Sameony) [inline src/frontend/style/cart.css:40]: you can create flex utility classes later on

**Review by Sameony — APPROVED** [review]
the code is simple and clean, loved the use of utility classes in there

File: src/frontend/context/cartContext.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        }).then(res => {
+            console.log(res.data);
+            setCart(res.data.cart)
+        }).catch((error) => {
+            // here you will have access to error.response
+            console.log(error.response.data)
+        })
+    }, [])
+
+    const decreaseQuantity = (product) => {
```
Comment (shubhamkumar648) [inline src/frontend/context/cartContext.js:71]: suggestion:- you can keep this function in utils folder for better readability , otherwise it is also fine

**Review by shubhamkumar648 — APPROVED** [review]
Remaining functionality is good

**Review by Priyanshi-Tiwari — APPROVED** [review]
just remove all the console logs other than that the code is clean and descriptive. great naming convention.

File: src/frontend/style/cart.css (hunk 3 above)
Comment (ZuberDunge) [reply src/frontend/style/cart.css:40]: I've created those classes, forgot to use, Will update.

File: src/frontend/context/cartContext.js (hunk 4 above)
Comment (ZuberDunge) [reply src/frontend/context/cartContext.js:71]: thank you for the feedback, I'll make the changes.

**Pull request merged**